frontend/lib/shared/number_pad.dart

```diff
@@ -74,10 +69,18 @@ class NumberPadKey extends HookWidget {
     return GestureDetector(
       onTapDown: (_) => keySize.value = selectedFontSize,
       onTapCancel: () => keySize.value = defaultFontSize,
-      onTapUp: (key) {
+      onTapUp: (_) {
         keySize.value = defaultFontSize;
-        (title == '<') ? onDeletePressed() : onKeyPressed(title);
+
+        title == '<'
+            ? _onDeletePressed(enteredAmount)
+            : (title == '.' && enteredAmount.value.contains('.'))
+                ? null
+                : !RegExp(r'\.\d{2}$').hasMatch(enteredAmount.value)
+                    ? _onKeyPressed(title, enteredAmount)
+                    : null;
```

Comment:
not really happy with how this looks, potentially change to if else blocks?

Reply:
I think maybe `if`s with early returns would look better here as well
```dart
if (title == '<') {
  //deal with title
  return;
}
```